src/util/notifications_util.js

```diff
@@ -494,10 +499,31 @@ function get_notification_logger(locking, namespace, poll_interval) {
         namespace = node_name + '_' + config.NOTIFICATION_LOG_NS;
     }
 
-    return new PersistentLogger(config.NOTIFICATION_LOG_DIR, namespace, {
+    const logger = new PersistentLogger(config.NOTIFICATION_LOG_DIR, namespace, {
         locking,
         poll_interval,
     });
+
+    //initialize writes_counter, used in check_free_space
+    logger.writes_counter = 0;
+
+    return logger;
+}
+
+//If space check is configures, create an event in case free space is below threshold.
+function check_free_space(req) {
+    if (!req.object_sdk.nsfs_config_root || !config.NOTIFICATION_REQ_PER_SPACE_CHECK) {
+        return;
+    }
+    req.notification_logger.writes_counter += 1;
+
+    if (req.notification_logger.writes_counter > config.NOTIFICATION_REQ_PER_SPACE_CHECK) {
+        req.notification_logger.writes_counter = 0;
+        const fs_stat = fs.statfsSync(config.NOTIFICATION_LOG_DIR);
+        if (fs_stat.bavail / fs_stat.blocks < config.NOTIFICATION_SPACE_CHECK_THRESHOLD) {
```

Comment:
Why at start do you init the `writes_counter` to 0?
Could you explain in a comment the logic for the calc?

Reply:
writes_counter counts how many writes has been performed since last check.
it starts at zero, and then zeroed on each check.